Applications/Spire/Source/Ui/QuantityInputWidget.cpp

```
@@ -1,46 +1,68 @@
 #include "Spire/Ui/QuantityInputWidget.hpp"
-#include <QHBoxLayout>
-#include <QRegularExpressionValidator>
 
 using namespace boost::signals2;
 using namespace Nexus;
 using namespace Spire;
 
-QuantityInputWidget::QuantityInputWidget(QWidget* parent)
-    : TextInputWidget(parent),
-      m_item_delegate(this) {
-  setValidator(new QRegularExpressionValidator(
-    QRegularExpression("^[0-9]*$"), this));
-  connect(this, &QLineEdit::editingFinished, this,
-    &QuantityInputWidget::on_line_edit_committed);
-  connect(this, &QLineEdit::textEdited, this,
-    &QuantityInputWidget::on_line_edit_modified);
-  m_locale.setNumberOptions(QLocale::OmitGroupSeparator);
+namespace {
+  auto real_to_quantity(NumericInputWidget::Real value) {
+    auto quantity = Quantity::FromValue(value.str(
+      std::numeric_limits<Quantity>::digits10, std::ios_base::dec));
+    if(quantity != boost::none) {
+      return *quantity;
+    }
+    return Quantity();
```

Don't silently fail.